lib/seahorn/BvOpSem2Allocators.cc

```diff
@@ -296,7 +298,23 @@ class StaticOpSemAllocator : public OpSemAllocator {
       preAlloc(inst, memSz, true);
     } else {
       // -- allocate max allowed for dynamically sized allocations
-      preAlloc(inst, m_maxSymbAllocSz, false);
+      // -- use inferred upper bound if crab analysis is performed
+      auto operand_inst = dyn_cast<llvm::Instruction>(inst.getOperand(0));
+      auto rangeValCrab = m_sem.getCrabInstRng(*operand_inst);
+      auto rangeValLVI = m_sem.getLVIInstRng(*operand_inst);
+      uint64_t ub_crab = rangeValCrab.getUpper().getLimitedValue();
+      uint64_t ub_lvi = rangeValLVI.getUpper().getLimitedValue();
```

Comment:
where is `ub_lvi` used?

Reply:
I haven't used that because the test I made shows LVI analysis is not precise as crab did. Maybe the current solution is only to choose the most precise result from either crab inferred or lvi inferred